nautobot_netbox_importer/diffsync/netbox.py

```diff
@@ -0,0 +1,505 @@
+"""NetBox to Nautobot importer."""
+import json
+from pathlib import Path
+from typing import Tuple
+from typing import Union
+
+from django.db.transaction import atomic
+from nautobot.ipam.models import get_default_namespace
+
+from nautobot_netbox_importer.diffsync.nautobot import NautobotAdapter
+
+from .base import EMPTY_VALUES
+from .base import RecordData
+from .source import SourceAdapter
+from .source import SourceField
+from .source import SourceRecord
+from .summary import print_fields_mapping
+from .summary import print_summary
+
+
+def _define_tagged_object(field: SourceField) -> None:
+    field.set_nautobot_field(field.name)
+
+    def importer(source: RecordData, target: RecordData) -> None:
+        content_type = source.get("content_type", None)
+        object_id = source.get(field.name, None)
+        if not object_id or not content_type:
+            target[field.name] = None
+            return
+
+        related_wrapper = field.wrapper.adapter.get_or_create_wrapper(content_type)
+        result = related_wrapper.get_pk_from_uid(object_id)
+        target[field.name] = result
+        related_wrapper.add_reference(result, field.wrapper)
+
+    field.set_importer(importer)
+
+
+def _define_choices(field: SourceField) -> None:
+    choices_wrapper = field.wrapper.adapter.get_or_create_wrapper("extras.customfieldchoice")
+
+    def importer(source: RecordData, target: RecordData) -> None:
+        choices = source.get(field.name, None)
+        if not choices:
+            return
+
+        choices = json.loads(choices)
+        if not choices:
+            return
+
+        for choice in choices:
+            choices_wrapper.import_record(
+                {
+                    "id": choice,
+                    "custom_field": target["id"],
+                    "value": choice,
+                },
+            )
+
+    field.set_importer(importer)
+
+
+def _name_from_wrapper(wrapper) -> str:
+    """Return model name with upper cased first letter.
+
+    e.g. `dcim.site` -> `Site`
```

Comment:
Any concern about cases where this isn't "correct", e.g. `VLAN`, `IPAddress`?

Reply:
Removed not to be confusing. It's used for location types only.